### DISCUSSION TITLE
How to use InflateRaw?

### DISCUSSION THREAD

**Discussion by mustafasalahuldin** [gildas-lormeau/zip.js#323] [Q&A]
I need to use InflateRaw same as in pako while using zip.js, is it possible?
I can read the entry from the zip file as Byte Array, what is pending is to inflate it and show the progress.
Please advise!

**Comment by gildas-lormeau** [comment]
You should use `Entry#getData()` to inflate data, see https://gildas-lormeau.github.io/zip.js/core-api.html#zip-reading.

**Reply by mustafasalahuldin** [reply]
This is not working in case I have the entry as Uint8Array.
In my case I use an HTTP request to extract the entry I need from the zip file on server, then I use pako.inflateRaw(Uint8Array) to unzip the entry(as binary), which works perfectly, the issue is pako doesn't provide the onProgress option.
So I need to do the same using zip.js, pass entry as Uint8Array then inflate it and monitor the progress.
Thanks in advance.

**Reply by mustafasalahuldin** [reply]
Just for more clarity, in my case, I'm inflate binary entry not the full zip. so there is no Central Directory.

**Reply by gildas-lormeau** [reply]
It's here: https://github.com/gildas-lormeau/zip.js/blob/master/lib/core/codecs/inflate.js. See the class `ZipInflate(options)` class at the end of the file, it has an `append` method and a `flush` method that allow you to implement a progress listener. Note that pako does have a progress listener but it looks like it's undocumented. It's the abstract method `onData` in the object returned by `new pako.Inflate()` and `new pako.Deflate()` (see last examples here: https://github.com/nodeca/pako#examples--api).

**Reply by mustafasalahuldin** [reply]
Thanks a lot. Would you please show an example for ZipInflate(options) using Typescript? I can't define such module.

**Reply by mustafasalahuldin** [reply]
@gildas-lormeau Also I'm interested in InflateRaw() as from the [documentation page](https://nodeca.github.io/pako/#inflateRaw), it seems there is no progress event assigned to inflate raw.
